packages/data-context/src/sources/BrowserDataSource.ts

```diff
@@ -33,6 +42,41 @@ export interface BrowserApiShape {
 export class BrowserDataSource {
   constructor (private ctx: DataContext) {}
 
+  /**
+   * Gets the browsers from the machine and the project config
+   */
+  allBrowsers () {
+    if (!this.ctx.coreData.allBrowsers) {
+      const p = this.ctx.project.getConfig()
+      const machineBrowsers = this.machineBrowsers()
+
+      this.ctx.coreData.allBrowsers = Promise.all([p, machineBrowsers]).then(async ([cfg, m]) => {
+        if (!cfg.browsers) return m
+
+        const userBrowsers = cfg.browsers.map<FoundBrowser | undefined>((b) => {
```

Comment:
Whenever I see a `map` then `filter` combo I like to opt for `reduce` to decrease the number of iterations. Unless you think that would be more difficult to read?

Reply:
Not a strong opinion. Some folks find reduce to be less readable. If there's a preference for reduce, I can make that change.